connector.go

```diff
@@ -139,7 +136,8 @@ func (c connector) pullImage(ctx context.Context, image string) error {
 		}
 		parts := strings.Split(image, ":")
 		tag := parts[len(parts)-1]
-		if len(parts) > 1 && tagRegexp.MatchString(tag) && tag != "latest" && imageExists {
+		// It appears that the tag check is to defer invalid tag errors to docker
+		if len(parts) > 1 && validTagRegexp.MatchString(tag) && tag != "latest" && imageExists {
 			return nil
 		}
```

Comment:
Do you think this deserves an issue, or do you think I should just remove that logic?

Reply:
I think that the behavior of the "If Not Present" policy should not depend on the image tag (at least, not if it's a legitimate tag for an image existing in a remote repository); therefore, I think the test for `tag != "latest"` should be removed from here.

Whether that should be done in this PR or not is up to you (I don't think that adding that change will overburden this PR).

If you agree that it should be removed but think it should be done in a separate PR, then an issue would be appropriate...but, instead of opening a new issue, you could (IMO) just post a PR for it....  🙃

Reply:
Updated.